### PR TITLE
Fix: set default git config search path for tests

### PR DESCRIPTION
**Pull request by weihanglo**
Fixes https://github.com/rust-lang/cargo/issues/8863 by setting the default config search path.

Just wait https://github.com/rust-lang/git2-rs/pull/656 being merged and update Cargo.toml the new release of git2-rs 😄

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! I don't think this is the correct solution, however, because each thread has its own unique search path, right? The libgit2 search path is a global configuration option so if all threads are always changing it then I don't think that it is guaranteed to be configured correctly for each test?

**Comment by weihanglo** [comment]
> Thanks for the PR! I don't think this is the correct solution, however, because each thread has its own unique search path, right? The libgit2 search path is a global configuration option so if all threads are always changing it then I don't think that it is guaranteed to be configured correctly for each test?

My fault. I didn't notice that 😅. The original goal is to prevent global configurations from affecting tests. What if we set the search path to nowhere (maybe a empty directory?) and let git configs keep the defaults?

**Comment by alexcrichton** [comment]
That seems reasonable to me, yeah, having a sort of one-time configuration which points it to a blank location.

File: crates/cargo-test-support/src/git.rs [hunk 1]
```diff
@@ -124,11 +124,31 @@ impl Repository {
...
+    default_search_path();
     let repo = t!(git2::Repository::init(path));
     default_repo_cfg(&repo);
     repo
 }
 
+fn default_search_path() {
+    use crate::paths::GLOBAL_ROOT;
+    use git2::{opts::set_search_path, ConfigLevel};
+    lazy_static::lazy_static! {
```
Comment (alexcrichton) [inline crates/cargo-test-support/src/git.rs:136]: FWIW instead of using `lazy_static` this can use a `sync::sync::Once`

Comment (weihanglo) [reply crates/cargo-test-support/src/git.rs]: Fixed. Thanks!

This pull request is almost ready for review. Is there any plan to rollout a new release of git2?

Comment (alexcrichton) [reply crates/cargo-test-support/src/git.rs]: Sure yeah, I just published a new version.

**Comment by weihanglo** [comment]
Since [git2-rs#0.13.16](https://github.com/rust-lang/git2-rs/releases/tag/0.13.16) is released. This pull request is now ready for review.

**Comment by alexcrichton** [comment]
Thanks! Could this bump the version requirement on `git2` to ensure it pulls in the required version as well? That'll help updates in existing projects like rust-lang/rust which already have a lock file pointing to the older version of git2

**Pull request merged**